src/creatwth.cpp

```diff
@@ -892,6 +911,146 @@ BOOL WINAPI DetourUpdateProcessWithDllEx(_In_ HANDLE hProcess,
         return FALSE;
     }
     return TRUE;
+}
+
+BOOL WINAPI DetourInjectProcessWithDll(_In_ HANDLE hProcess,
+                                       _In_ LPCSTR pszDllName,
+                                       _In_ DWORD dwOrdinal)
+{
+    LPCSTR rlpDlls[2];
+    DWORD rdwOrdinals[2];
+    DWORD nDlls = 0;
+
+    if (pszDllName != NULL) {
+        rdwOrdinals[nDlls] = dwOrdinal;
+        rlpDlls[nDlls++] = pszDllName;
+    }
+
+    return DetourUpdateProcessWithDllWithOrdinals(hProcess, rlpDlls, rdwOrdinals, nDlls);
+}
+
+BOOL WINAPI DetourUpdateDllWithDll(_In_ HANDLE hProcess,

```

Comment:
I haven't read this closely, but is "so much" new/copied code required, or can it be refactored?